### DISCUSSION TITLE
Issues with PostCSS

### DISCUSSION THREAD

**Discussion by jamie-moxon** [thuliteio/doks#787] [Q&A]
Hi all, 

I am running into an issue when running the "npm run build" command for building my prod environment locally. It seems to be an issue with PostCSS:

```
Start building sites … 
hugo v0.99.0-1de333e7a3fc863672ec6d6cd53ba66dbcdd2305+extended windows/amd64 BuildDate=2022-05-16T08:10:56Z VendorInfo=gohugoio
Error: Error building site: POSTCSS: failed to transform "main.css" (text/css): 'C:\Program' is not recognized as an internal or external command,
operable program or batch file.
Total in 780 ms
```

Have anyone else had this issue? Using VS code on Win11.

Thanks

Jamie

**Comment by h-enk** [comment]
What versions of node + npm are you running? Do you have a repo you could share?

**Reply by h-enk** [reply]
Could you please paste the output of `npm list`?

**Reply by jamie-moxon** [reply]
Hi, Thanks for responding!

**Node v:** 16.15.1
**NPM v:** 8.11.0

Here is npm list:

npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.
doks-child-theme@0.4.3 
├── @azure/static-web-apps-cli@1.0.1
├── @babel/cli@7.17.10
├── @babel/core@7.17.12
├── @babel/preset-env@7.17.12
├── @fullhuman/postcss-purgecss@4.1.3
├── @hyas/doks@0.4.3
├── auto-changelog@2.4.0
├── autoprefixer@10.4.7
├── bootstrap@5.1.3
├── clipboard@2.0.11
├── eslint@8.15.0
├── exec-bin@1.0.0
├── flexsearch@0.7.21
├── highlight.js@11.5.1
├── hugo-installer@3.1.0
├── instant.page@5.1.0
├── katex@0.15.3
├── lazysizes@5.3.2
├── markdownlint-cli2@0.4.0
├── postcss-cli@9.1.0
├── postcss@8.4.14
├── purgecss-whitelister@2.4.0
├── shx@0.3.4
├── stylelint-config-standard-scss@3.0.0
└── stylelint@14.8.2


My Github repo is: https://github.com/moccolfc/GCIS

Thanks

Jamie

**Reply by h-enk** [reply]
> My Github repo is: https://github.com/moccolfc/GCIS

That helps /clarifies! You've removed a substantial number of essential settings and files (at least for production). Instead of trying to fix your repo (it would take too much time, and leave things unstable), I suggest you start all over (and leave things in, they're not hurting).

**Reply by jamie-moxon** [reply]
Ok, I may look at starting again then from scratch, Thanks again,

Jamie

**Reply by h-enk** [reply]
OK, sounds goods. Note that, maybe there are some specifics when hosting on Azure (see also https://github.com/h-enk/doks/discussions/657). If you got it running, it would be interesting to share here how!!

**Comment by jamie-moxon** [comment]
So I forked the repo https://github.com/h-enk/doks

and I still run into the same issue with PostCSS. I can run "npm run start" and the site will load locally, soon as I try to build, I still have the same error. 

npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.

> @hyas/doks@0.4.3 prebuild
> npm run clean

npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.

> @hyas/doks@0.4.3 clean
> shx rm -rf public resources


> @hyas/doks@0.4.3 build
> exec-bin node_modules/.bin/hugo/hugo --gc --minify

Start building sites … 
hugo v0.99.0-1de333e7a3fc863672ec6d6cd53ba66dbcdd2305+extended windows/amd64 BuildDate=2022-05-16T08:10:56Z VendorInfo=gohugoio
Error: Error building site: POSTCSS: failed to transform "main.css" (text/css): 'C:\Program' is not recognized as an internal or external command,
operable program or batch file.
Total in 801 ms

**Reply by roheclassic** [reply]
Is it possible to change almost all index.md files to draft(```draft: true```) for a moment?
If that makes the event unreproduced, then one of them has a problem.

**Reply by h-enk** [reply]
Looks like a thing with the combination Windows 11 and npm 8.11 — see also https://stackoverflow.com/a/72597304

Could you try downgrading npm to 8.10 (or lower)?

**Reply by jamie-moxon** [reply]
Tried downgrading to 8.10 and other lower versions and still running in to the same issues. Ive tried some of the fixes on that stackoverflow thread and no luck, Ill try on a Win10 device.
